snapcraft/meta/appstream.py

```diff
@@ -100,6 +100,13 @@ def extract(relpath: str, *, workdir: str) -> Optional[ExtractedMetadata]:
     description = _get_value_from_xml_element(dom, "description")
     title = _get_value_from_xml_element(dom, "name")
     version = _get_latest_release_from_nodes(dom.findall("releases/release"))
+    project_license = _get_value_from_xml_element(dom, "project_license")
+    contact = _get_value_from_xml_element(dom, "update_contact")
+
+    issues = _get_urls_from_xml_element(dom.findall("url"), "bugtracker")
+    donation = _get_urls_from_xml_element(dom.findall("url"), "donation")
+    website = _get_urls_from_xml_element(dom.findall("url"), "homepage")
+    source_code = _get_source_code_from_xml_element(dom.findall("url"))
```

Comment:
This seems like a special case of `_get_urls_from_xml_element`?

```suggestion
    source_links = _get_urls_from_xml_element(dom.findall("url"), "vcs-browser")
    source_code = source_links[0] if source_links else None
```

Reply:
According to the docs of snapcraft `source-code` is the only one that can be a `string`. It can't be a `List[str]`. So, if I do return it from the same function, it'll return a list, that'd change the definition in the docs. Should I do that?

Reply:
That's what the second line in my suggestion does - takes the one item from `source-links` and turns it into a string.

It's not a big deal really though, and I'm fine to leave it as-is.